google_guest_agent/cfg/cfg_test.go

```diff
@@ -0,0 +1,92 @@
+package cfg
+
+import "testing"
+
+func TestLoad(t *testing.T) {
+	if err := Load(); err != nil {
+		t.Fatalf("Failed to load configuration: %+v", err)
+	}
+
+	cfg, err := Get()
+	if err != nil {
+		t.Fatalf("Failed to retrieve configuration instance: %+v", err)
+	}
+
+	if cfg.WSFC != nil {
+		t.Errorf("WSFC shouldn't not be defined by default configuration, expected: nil, got: non-nil")
+	}
+
+	if cfg.AccountManager.Disable == true {
+		t.Errorf("Expected AccountManager.disable to be: false, got: true")
+	}
+
+	if cfg.Accounts.DeprovisionRemove == true {
+		t.Errorf("Expected Accounts.deprovision_remove to be: false, got: true")
+	}
+}
+
+func TestInvalidConfig(t *testing.T) {
+	invalidConfig := `
+[Section
+key = value
+`
+
+	dataSources = func() []interface{} {
+		return []interface{}{
+			[]byte(invalidConfig),
+		}
+	}
+
+	if err := Load(); err == nil {
+		t.Errorf("Load() didn't fail to load invalid configuration, expected error")
+	}
+
+	dataSources = defaultDataSources
```

Comment:
you can reset it with defer where its overridden on line 34 `defer dataSources = defaultDataSources`

Reply:
Done. thanks.